pkg/config/config.go

```diff
@@ -291,134 +77,87 @@ func InitCliConfig(configAndStacksInfo schema.ConfigAndStacksInfo, processStacks
-			if err != nil {
-				return atmosConfig, err
-			}
-			errorMessage := fmt.Sprintf("\nno stack manifests found in the provided "+
-				"paths:\n%s\n\nCheck if `base_path`, 'stacks.base_path', 'stacks.included_paths' and 'stacks.excluded_paths' are correctly set in CLI config "+
-				"files or ENV vars.", j)
-			return atmosConfig, errors.New(errorMessage)
-		}
-
-		atmosConfig.StackConfigFilesAbsolutePaths = stackConfigFilesAbsolutePaths
-		atmosConfig.StackConfigFilesRelativePaths = stackConfigFilesRelativePaths
-
-		if stackIsPhysicalPath {
-			u.LogTrace(fmt.Sprintf("\nThe stack '%s' matches the stack manifest %s\n",
-				configAndStacksInfo.Stack,
-				stackConfigFilesRelativePaths[0]),
-			)
-			atmosConfig.StackType = "Directory"
-		} else {
-			// The stack is a logical name
-			atmosConfig.StackType = "Logical"
-		}
-	}
-
-	atmosConfig.Initialized = true
-	return atmosConfig, nil
+	return nil
 }
 
-// https://github.com/NCAR/go-figure
-// https://github.com/spf13/viper/issues/181
-// https://medium.com/@bnprashanth256/reading-configuration-files-and-environment-variables-in-go-golang-c2607f912b63
-func processConfigFile(
-	atmosConfig schema.AtmosConfiguration,
-	path string,
-	v *viper.Viper,
-) (bool, error) {
-	// Check if the config file exists
-	configPath, fileExists := u.SearchConfigFile(path)
-	if !fileExists {
-		return false, nil
-	}
-
-	reader, err := os.Open(configPath)
+func processStackConfigs(atmosConfig *schema.AtmosConfiguration, configAndStacksInfo *schema.ConfigAndStacksInfo, includeStackAbsPaths, excludeStackAbsPaths []string) error {
+	// If the specified stack name is a logical name, find all stack manifests in the provided paths
+	stackConfigFilesAbsolutePaths, stackConfigFilesRelativePaths, stackIsPhysicalPath, err := FindAllStackConfigsInPathsForStack(
+		*atmosConfig,
+		configAndStacksInfo.Stack,
+		includeStackAbsPaths,
+		excludeStackAbsPaths,
+	)
 	if err != nil {
-		return false, err
+		return err
 	}
 
-	defer func(reader *os.File) {
-		err := reader.Close()
+	if len(stackConfigFilesAbsolutePaths) < 1 {
+		j, err := u.ConvertToYAML(includeStackAbsPaths)
 		if err != nil {
-			u.LogWarning(fmt.Sprintf("error closing file '%s'. %v", configPath, err))
+			return err
 		}
-	}(reader)
+		errorMessage := fmt.Sprintf("\nno stack manifests found in the provided "+
+			"paths:\n%s\n\nCheck if `base_path`, 'stacks.base_path', 'stacks.included_paths' and 'stacks.excluded_paths' are correctly set in CLI config "+
+			"files or ENV vars.", j)
+		return errors.New(errorMessage)
+	}
 
-	err = v.MergeConfig(reader)
-	if err != nil {
-		return false, err
+	atmosConfig.StackConfigFilesAbsolutePaths = stackConfigFilesAbsolutePaths
+	atmosConfig.StackConfigFilesRelativePaths = stackConfigFilesRelativePaths
+
+	if stackIsPhysicalPath {
+		log.Debug(fmt.Sprintf("\nThe stack '%s' matches the stack manifest %s\n",
+			configAndStacksInfo.Stack,
+			stackConfigFilesRelativePaths[0]))
+		atmosConfig.StackType = "Directory"
```

Comment:
Not sure if this makes sense to me. What is a "directory" stack type?

Cc @aknysh